ksqldb-engine/src/test/java/io/confluent/ksql/engine/KsqlEngineTest.java

```diff
@@ -347,6 +347,31 @@ public void shouldThrowWhenInsertIntoSchemaDoesNotMatch() {
         is("insert into bar select orderTime, itemid from orders;")));
   }
 
+  @Test
+  public void shouldExecuteInsertIntoWithCustomQueryId() {
```

Comment:
Can we add a test in `RecoveryTest` to make sure we can recover create/terminate with custom query IDs?

Reply:
Done